airbyte-cdk/python/airbyte_cdk/sources/abstract_source.py

```diff
@@ -106,6 +107,11 @@ def read(
                         f"The requested stream {configured_stream.stream.name} was not found in the source."
                         f" Available streams: {stream_instances.keys()}"
                     )
+                if self.availability_strategy is not None:
+                    stream_is_available, error = self.availability_strategy.check_availability(stream_instance)
+                    if not stream_is_available:
+                        logger.info(f"Skipped syncing stream '{stream_instance.name}' because it was unavailable. Error: {error}")
```

Comment:
@evantahler what was it you said about info we want in the error message? Also is this only for logging to the stream sync info, or should we output an AirbyteConnectionStatusMessage or something similar?